#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()
gen.add("min_index", int_t, 0, "the minimum index to filter", 0, 0, 4096)
gen.add("max_index", int_t, 0, "the maximum index to filter", 640, 0, 4096)
gen.add("keep_organized", bool_t, 0, "keep organized pointcloud or not", True)
gen.add("remove_nan", bool_t, 0, "remove NaN point from pointcloud", False)
gen.add("filter_limit_negative", bool_t, 0,
        "toggle the points which are listed or not", False)

field_enum = gen.enum([gen.const("X", int_t, 0, "Filter based on x field"),
                       gen.const("Y", int_t, 1, "Filter based on y field")],
                       "filter field")
gen.add("filter_field", int_t, 0, "filter field", 0, 0, 1,
        edit_method = field_enum)
exit (gen.generate (PACKAGE, "jsk_pcl_ros", "OrganizedPassThrough"))
